### ISSUE TITLE
Define PrimaryPurpose field

### ISSUE THREAD

**Issue by dturnerx** [KantaraInitiative/CISWG#89]
PrimaryPurpose field: This is an undefined term. It is unclear how this field should be used. (On a legal point: under data minimisation principles, should data be collected for secondary purposes? I suspect that the concept of a hierarchy of purposes would not hold much water.)

**Comment by PrivacyCDN** [comment]
I disagree. There is a hierarchy of purposes. Some data is collected inevitably because of the technology (browser header info for example). This is collected on landing. Some data will be collected because it is necessary to provide a service or because of regulation (booking a flight requires the collect of PNR data for ticket issuance & international security requirements, credit card info for payment and so on). This is collected, presumably after notice. Finally, there is data collection or use that is optional (can we send your marketing, or do you want to fill out a survey. This last will require an opt-in or affirmative button click. That's at least three tiers of collection with differing notice/consent expectations.

I note that all of this is a separate issue with the phrase (secondary use). In health for example that often refers to health data used for health research that may be collected regardless of initial consent and notice for research purposes.

**Comment by andrewhughes3000** [comment]
@PrivacyCDN any suggestions on a path forward to resolve the issue?

**Comment by PrivacyCDN** [comment]
My first inclination is to suggest that a data model would provide a toolset for specifying purposes and their characteristics. If the articulation of a data model is put off to v2, we may be able to do it another way, in prose and diagrams.  If we weren't virtual and had access to a room I'd suggest an old fashioned affinity diagram bashing session.

**Comment by smartopian** [comment]
Seems that guidance is needed here, but we purposely kept this a bit vague so we had room to define how to specify purpose.  The shining is that if its not a core purpose/primary purpose then its a secondary purpose.  This is important because there is a requirement for people to be able to withdraw secondary purposes in context but not withdraw the entire consent. 

But, we haven’t thought through what that means for multiple purposes.  (Not sure we need to) arguably to know if this works we need to nail down the receipt taxonomy.   i.e. each service can have multiple purpose categories  and then does the primary purpose (or not) field sit at the category level (which we have now) -  personally I think its correct as is -but - maybe an affinity diagram will help?

**Comment by linnhege** [comment]
We use the PrimaryPurpose field to separate between consent that is to given up front (login, account creation) and consent that is given contextually. Just my two cents :)

**Comment by dturnerx** [comment]
@linnhege - can you elaborate on what you mean by "consent that is given contextually". Specifically, what does this look like in practice?

**Comment by PrivacyCDN** [comment]
And one last thing to remind people. That is the term 'Secondary Use'. In my experience, this is NOT related to secondary purposes. It is most usually used in a medical setting to access and use data for research that was not collected with a research purpose. My suggested buckets of purposes are:

Bucket 1: Technical necessity. The sole purpose here is to collect/user information for the necessary functioning of a system. e.g. Browser Header Information

Bucket 2: Primary service. The purpose(s) that identify the minimum information that can be collected/used to fulfill the requirements of the service on offer. e.g. User name and password to register on a site

Bucket 3: Other services (to avoid the confusing 'secondary'). The purpose(s) that identify information requested but not required for the service on offer. e.g. a consent directive to allow secondary marketing to the user.

Each of the above buckets could contain a list of 'standard' purposes.

John Wunderlich,

Sent frum a mobile device,
Pleez 4give speling erurz

"...a world of near-total surveillance and endless record-keeping is likely to be one with less liberty, less experimentation, and certainly far less joy..." A. Michael Froomkin
________________________________

**Comment by maryhodder** [comment]
Bucket 3 should also include 'derived' data which is part of the gdpr
requirements and if a company derives something for a purpose or to a
conclusion, depending, it can have consequences for purposes as well as
other categories such as whether something is sensitive personal data.

**Comment by PrivacyCDN** [comment]
See my recent comment on issue #96 included consent for core services vs optional consents. Is Primary Purpose functional equivalent to 'those purposes that are minimally required to deliver the core service'?

**Comment by RupertGravesDIC** [comment]
If the service requires the purpose then it requires the purpose, whether it is minimally or not. Hence all purposes work in parallel (aside from some niche cases e.g. healthcare research). The underlying issue is whether purpose scope (intentionally) expands beyond what is necessary to provide the service, i.e. purpose creep. I would suggest that this can be tested normatively using ISO 29100 data privacy principles by considering whether a hypothesis definiton would change in different privacy contexts.
